cdap-data-fabric/src/test/java/co/cask/cdap/api/dataset/lib/KeyValueTableTest.java

```diff
@@ -223,9 +223,6 @@ public void apply() throws Exception {
       Assert.assertEquals("Cancel transaction", e.getCause().getMessage());
     }
 
-    // add a swap for a third table that should fail
-    Assert.assertFalse(kvTable.compareAndSwap(KEY3, VAL1, VAL1));
```

Comment:
The idea of this, I believe, was to fail the transaction and then test the rollback. This was a left-over from an older version of datasets where compareAndSwap failed the transaction instead of returning false. The transaction gets rolled back by the Assert.fail a couple of lines before, and this serves no real purpose. But I can bring it back, it can run in its own tx and test the compareAndSwap.